control-plane/agents/src/bin/core/controller/reconciler/nexus/mod.rs

```diff
@@ -130,49 +134,247 @@ async fn nexus_reconciler(
     }
 }
 
-/// Find and removes faulted children from the given nexus
-/// If the child is a replica it also disowns and destroys it
+/// Checks if nexus is Degraded and any child is Faulted. If yes, Depending on rebuild policy for
+/// child sets the rebuild stage in hashmap to drive the rebuild. We exclude NoSpace Degrade.
 #[tracing::instrument(skip(nexus, context), level = "trace", fields(nexus.uuid = %nexus.uuid(), request.reconcile = true))]
-pub(super) async fn faulted_children_remover(
+pub(super) async fn handle_faulted_child(
     nexus: &mut OperationGuardArc<NexusSpec>,
     context: &PollContext,
 ) -> PollResult {
     let nexus_uuid = nexus.uuid();
     let nexus_state = context.registry().nexus(nexus_uuid).await?;
     let child_count = nexus_state.children.len();
+    if nexus_state.status == NexusStatus::Degraded && child_count > 1 {
+        let nexus_spec_clone = nexus.lock().clone();
+        for child in nexus_state.children.iter().filter(|c| c.state.faulted()) {
+            if !nexus_spec_clone.rebuild_state.contains_key(&child.uri)
+                && child.state_reason.clone() != ChildStateReason::NoSpace
+            {
+                let wait_time = RuleSet::faulted_child_wait(&nexus_state);
+                if wait_time.is_zero() {
+                    info!("Start full rebuild");
```

Comment:
What if the replica is not online?

Reply:
This will remove the child from children list to trigger Full rebuild. mostly in scenario when 2nd child gets Faulted in 3 replica vilume. First will be in TImedWait. 2nd will be removed in FullRebuildInit stage